docker/docker_client.go

```diff
@@ -212,37 +208,48 @@ func hasFile(files []os.FileInfo, name string) bool {
 	return false
 }
 
-// newDockerClient returns a new dockerClient instance for refHostname (a host a specified in the Docker image reference, not canonicalized to dockerRegistry)
-// “write” specifies whether the client will be used for "write" access (in particular passed to lookaside.go:toplevelFromSection)
-func newDockerClient(ctx *types.SystemContext, ref dockerReference, write bool, actions string) (*dockerClient, error) {
+func newDockerClientFromRef(ctx *types.SystemContext, ref dockerReference, write bool, actions string) (*dockerClient, error) {
 	registry := reference.Domain(ref.ref)
-	if registry == dockerHostname {
-		registry = dockerRegistry
-	}
```

Comment:
I can’t see where this code was migrated to; am I overlooking anything?

(If not, this mapping should probably happen in `newDockerClientWithDetails`, so that the callers of `reference.Domain(ref.ref)` can use the original `registry` parameter, and the users of the modified value can use some other variable.)

Reply:
I thought we wanted this mapping to happen in `docker/config/config.go` so that is where I moved it to.

Reply:
That takes care of reading the configuration, but not actually connecting over HTTPS.  (`CheckAuth` does this mapping immediately, so that works, but I can’t see that in the `newDockerClientFromRef` path; AFAICS that path will try to connect to the hostname `docker.io`, and fail)

Reply:
fixed